pokedex-with-bit/app.py

```diff
@@ -20,39 +23,50 @@ def config():
     os.environ['JINA_PORT'] = os.environ.get('JINA_PORT', str(45678))
 
 
-# for index
-def index():
+def index(num_docs: int):
     f = Flow.load_config('flows/index.yml')
+    num_docs = min(num_docs, len(glob(os.path.join(os.getcwd(), IMAGE_SRC), recursive=True)))
 
     with f:
-        f.index_files(image_src, request_size=64, read_mode='rb', size=num_docs)
+        with TimeContext(f'QPS: indexing {num_docs}', logger=f.logger):
+            f.index_files(IMAGE_SRC, request_size=64, read_mode='rb', size=num_docs)
 
 
-# for search
-def search():
+def query():
     f = Flow.load_config('flows/query.yml')
+    f.use_rest_gateway()
 
+    # no perf measure, as it opens a REST api and blocks
     with f:
         f.block()
 
 
-if __name__ == '__main__':
-    if len(sys.argv) < 2:
-        print('choose between "index" and "search" mode')
-        exit(1)
-    if sys.argv[1] == 'index':
-        config()
-        workspace = os.environ['JINA_WORKSPACE']
+@click.command()
+@click.option(
+    "--task",
+    "-t",
+    type=click.Choice(
+        ["index", "query"], case_sensitive=False
+    ),
+)
+@click.option("--num_docs", "-n", default=MAX_DOCS)
+def main(task: str, num_docs: int):
+    config()
+    workspace = os.environ["JINA_WORKSPACE"]
+    if task == "index":
         if os.path.exists(workspace):
-            print(f'\n +---------------------------------------------------------------------------------+ \
-                    \n |                                   🤖🤖🤖                                        | \
-                    \n | The directory {workspace} already exists. Please remove it before indexing again. | \
-                    \n |                                   🤖🤖🤖                                        | \
-                    \n +---------------------------------------------------------------------------------+')
-            sys.exit()
-        index()
-    elif sys.argv[1] == 'search':
-        config()
-        search()
-    else:
-        raise NotImplementedError(f'unsupported mode {sys.argv[1]}')
+            print(f'\n +----------------------------------------------------------------------------------+ \
+                    \n |                                   🤖🤖🤖                                         | \
+                    \n | The directory {workspace} already exists. Please remove it before indexing again.  | \
+                    \n |                                   🤖🤖🤖                                         | \
+                    \n +----------------------------------------------------------------------------------+')
+            sys.exit(1)
+        index(num_docs)
+    if task == "query":
+        if not os.path.exists(workspace):
+            print(f"The directory {workspace} does not exist. Please index first via `python app.py -t index`")
```

Comment:
add a `sys.exit(1)` here too, else it will still query

Reply:
See https://github.com/jina-ai/examples/pull/460/commits/39611cc68ff0fd28adadd996d794c3ea7b30da17